pkg/controller/cronjob/metrics/metrics.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const CronJobControllerSubsystem = "cronjob_controller"
+
+var (
+	CronJobCreationSkew = metrics.NewHistogram(
+		&metrics.HistogramOpts{
+			Subsystem:      CronJobControllerSubsystem,
+			Name:           "cronjob_job_creation_skew_duration_seconds",
+			Help:           "Time between when a cronjob is scheduled to be run, and when the corresponding job is created",
+			StabilityLevel: metrics.ALPHA,
+			Buckets:        []float64{1, 2.5, 5, 10, 15, 25, 50, 120, 300, 600, 1200},
+		},
+	)
+	CronJobMissedSchedulesTotal = metrics.NewGaugeVec(
+		&metrics.GaugeOpts{
+			Subsystem:      CronJobControllerSubsystem,
+			Name:           "cronjob_missed_schedules",
```

Comment:
please add the `_total` suffix to this (from https://prometheus.io/docs/practices/naming/#metric-names)

Reply:
Ah, it looks like this is the _current_ number of missed schedules, so the total suffix isn't required, so you can ignore the previous comment.  I'd suggest `cronjob_current_missed_schedules`.

Reply:
I was looking at the naming conventions and found this `node_memory_usage_bytes`.  IMO, `schedules` sounds like the right unit for this metric. Considering this if you still feel otherwise, please let me know I can append `_total` to it. @dashpole

Reply:
You are correct.  schedules is the correct unit.  You can optionally add `_current_` to the name if you want to make it clearer that it is an instantaneous, rather than cumulative metric.

Reply:
I don't think it is going to be instantaneous. I think schedules will be missed if the controller is overloaded or dead (not running for some reason). The controller will only learn about missed schedules after the fact, in retrospect, once it has recovered.

Reply:
By instantaneous, I just mean not-cumulative.  A "point-in-time" measurement, rather than a "sum-over-time" measurement.

Reply:
@alaypatel07 I'm assuming you're taking that other metrics as a followup?

Reply:
@soltysh yes, I plan to follow up.